File: June/27-06-2024/Toeplitz matrix.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

bool isToeplitz(vector<vector<int>> &mat);

int main() {
    // your code goes here
    int t;
    cin >> t;
    while (t--) {
        int n, m;
        cin >> n >> m;
        vector<vector<int>> a(n, vector<int>(m, 0));

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++)
                cin >> a[i][j];
        }

        bool b = isToeplitz(a);

        if (b == true)
            cout << "true";
        else
            cout << "false";

        cout << endl;
    }
    return 0;
}
// } Driver Code Ends



bool isToeplitz(vector<vector<int>>& mat) {
    //Your Code Start
      int r = mat.size();
    int c = mat[0].size();
    
    for (int row = 0; row < r - 1; ++row) {
        for (int col = 0; col < c - 1; ++col) {
            if (mat[row][col] != mat[row + 1][col + 1]) {
                return false;
            }
        }
    }
    return true;
  //Your Code End
}

File: June/27-06-2024/Toeplitz matrix.py
# You are required to complete this method
# Return True/False or 1/0
def isToeplitz(mat):
  #Your Code Start
    rows = len(mat)
    cols = len(mat[0])
    
    # Check if each element is equal to the element diagonally down-right
    for row in range(rows - 1):
        for col in range(cols - 1):
            if mat[row][col] != mat[row + 1][col + 1]:
                return False
    return True
  #Your Code End

#{ 
 # Driver Code Starts
# Your code goes here
if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        n, m = list(map(int, input().strip().split()))
        arr = list(map(int, input().strip().split()))
        matrix = [[0 for i in range(m)] for j in range(n)]
        k = 0
        for i in range(n):
            for j in range(m):
                matrix[i][j] = arr[k]
                k += 1
        b = isToeplitz(matrix)

        if b == True:
            print("true")
        else:
            print("false")

# } Driver Code Ends
